bench/bm_case1.cpp

```diff
@@ -7,44 +6,128 @@
 
 #include "bm_test_helper.hpp"
 #include <graph.hpp>
+#include <vir/simd.h>
 
-// #define RUN_SIMD_TESTS
+namespace fg = fair::graph;
 
-namespace fg                           = fair::graph;
-
-inline constexpr std::size_t N_ITER    = 10;
+inline constexpr std::size_t N_ITER = 10;
 inline constexpr std::size_t N_SAMPLES = gr::util::round_up(1'000'000, 1024);
 
 template<typename T, int addend, std::size_t N_MIN = 0, std::size_t N_MAX = N_MAX>
-class add : public fg::node<add<T, addend, N_MIN, N_MAX>, fg::IN<T, "in", N_MIN, N_MAX>, fg::OUT<T, "out", N_MIN, N_MAX>, fg::limits<N_MIN, N_MAX>> {
+class add
+        : public fg::node<add<T, addend, N_MIN, N_MAX>, fg::IN<T, "in", N_MIN, N_MAX>, fg::OUT<T, "out", N_MIN, N_MAX>, fg::limits<N_MIN, N_MAX>> {
 public:
     template<fair::meta::t_or_simd<T> V>
-    [[nodiscard]] constexpr T
+    [[nodiscard]] constexpr V
     process_one(V a) const noexcept {
         return a + addend;
     }
 };
 
 template<typename T, std::size_t N_MIN = 0, std::size_t N_MAX = N_MAX>
-class multiply : public fg::node<multiply<T, N_MIN, N_MAX>, fg::IN<T, "in", N_MIN, N_MAX>, fg::OUT<T, "out", N_MIN, N_MAX>, fg::limits<N_MIN, N_MAX>> {
+class multiply
+        : public fg::node<multiply<T, N_MIN, N_MAX>, fg::IN<T, "in", N_MIN, N_MAX>, fg::OUT<T, "out", N_MIN, N_MAX>, fg::limits<N_MIN, N_MAX>> {
     T _factor = static_cast<T>(1.0f);
 
 public:
     multiply() = delete;
 
-    explicit multiply(T factor, std::string name = fair::graph::this_source_location()) : _factor(factor) {
-        this->set_name(name);
-    }
+    explicit multiply(T factor, std::string name = fair::graph::this_source_location()) : _factor(
+            factor) { this->set_name(name); }
```

Comment:
did clang-format do this newline?

Reply:
unfortunately, yes ... :-(

we need to work on a more practical clang-format definition (-> one of the next PRs)